readthedocs/subscriptions/event_handlers.py

```diff
@@ -74,26 +118,39 @@ def _update_subscription_from_stripe(rtd_subscription, stripe_subscription_id):
             "Trial ended, canceling subscription.",
         )
         cancel_stripe_subscription(stripe_subscription.id)
+        return
 
+    organization = getattr(stripe_subscription.customer, "rtd_organization", None)
+    if not organization:
+        log.error("Subscription isn't attached to an organization")
+        return
 
-@handler("customer.subscription.updated", "customer.subscription.deleted")
-def update_subscription(event):
-    """Update the RTD subscription object with the updates from the Stripe subscription."""
-    stripe_subscription_id = event.data["object"]["id"]
-    rtd_subscription = Subscription.objects.filter(
-        stripe_id=stripe_subscription_id
-    ).first()
-    if not rtd_subscription:
+    save_org = False
```

Comment:
I'd like to have a comment here explaining in what cases we need to save the organization and in what cases we don't. This will help to make this code more clear.